main/lsp/requests/document_symbol.cc

```diff
@@ -63,17 +74,50 @@ std::unique_ptr<DocumentSymbol> symbolRef2DocumentSymbol(const core::GlobalState
     result->detail = "";
 
     vector<unique_ptr<DocumentSymbol>> children;
-    symbolRef2DocumentSymbolWalkMembers(gs, symRef, filter, children);
+    symbolRef2DocumentSymbolWalkMembers(gs, symRef, filter, defMapping, children);
     if (symRef.isClassOrModule()) {
         auto singleton = symRef.asClassOrModuleRef().data(gs)->lookupSingletonClass(gs);
         if (singleton.exists()) {
-            symbolRef2DocumentSymbolWalkMembers(gs, singleton, filter, children);
+            symbolRef2DocumentSymbolWalkMembers(gs, singleton, filter, defMapping, children);
         }
     }
     result->children = move(children);
     return result;
 }
 
+// Document symbols in the LSP spec include two different ranges: one that covers
+// the entire extent of the definition and one that covers just what should be
+// shown when the symbol is selected.  Sorbet's loc for a symbol covers the latter.
+// We do store the former, but that loc lives in the AST, not in the symbol table,
+// so we build a separate temporary mapping with this tree walker.
+class DefLocSaver {
```

Comment:
slight preference for a different name, because there is already `main/lsp/DefLocSaver.h` which has a separate purpose.